File: src/VisualStudio/Reqnroll.IdeSupport.VisualStudio.Extension/TestTargets/ScenarioTestTarget.cs
#nullable enable

namespace Reqnroll.IdeSupport.VisualStudio.Extension.TestTargets;

/// <summary>
/// One generated test method (or one row of a row-tests-parameterized method) that a
/// <c>.feature</c> scenario/Scenario Outline resolves to. Mirrors <c>ScenarioTestTargetDto</c> in
/// <c>ResolveTestTargetsResponse.cs</c> on the server (design doc §3/§4, issue #262).
/// </summary>
public sealed record ScenarioTestTarget(
    string DeclaringTypeFullName,
    string MethodName,
    bool IsParameterized,
    int? RowIndex);

File: src/VSCode/src/testRunner/scenarioTestTarget.ts
/**
 * Mirrors `ScenarioTestTargetDto`/`ResolveTestTargetsResponse` in
 * `src/LSP/Reqnroll.IdeSupport.LSP.Server/Features/TestTargets/ResolveTestTargetsResponse.cs` —
 * the response shape of the `reqnroll/resolveTestTargets` request (design doc §3/§4).
 */
export interface ScenarioTestTargetDto {
  readonly declaringTypeFullName: string;
  readonly methodName: string;
  readonly isParameterized: boolean;
  readonly rowArguments?: Record<string, string> | null;
  readonly rowIndex?: number | null;
}

export interface ResolveTestTargetsResponse {
  readonly targets: ScenarioTestTargetDto[];
}
